router/image/templates/nginx.conf

```diff
@@ -137,38 +144,38 @@ http {
         {{ end }}
     }
     ## end deis-store-gateway
-
+    {{ $useSSL := "false" }}
+    {{ if exists "/deis/router/sslCert" }}{{ $useSSL := getv "/deis/router/sslCert" }}{{ end }}
+    {{ $domains := ls "/deis/domains" }}
+    {{ $certs := ls "/deis/certs" }}
     ## start service definitions for each application
-    {{ $affinityArg := .deis_router_affinityArg }}
-    {{ $certs := .deis_certs }}
-    {{ $domains := .deis_domains }}
-    {{ range $service := .deis_services }}{{ if $service.Nodes }}
-    upstream {{ Base $service.Key }} {
-        {{ if $affinityArg }}hash $arg_{{ $affinityArg }} consistent;
+    {{ range $app := lsdir "/deis/services" }}
+    {{ $upstreams := printf "/deis/services/%s/*" $app}}
+    upstream {{ $app }} {
+        {{ if exists "/deis/router/affinityArg" }}
+        hash $arg_{{ getv "/deis/router/affinityArg" }} consistent;
         {{ end }}
-        {{ range $upstream := $service.Nodes }}server {{ $upstream.Value }};
+        {{range gets $upstreams }}server {{.Value}};
```

Comment:
This line should add back the whitespace for consistency with the rest of the `confd` template IMHO.